packages/task/src/browser/provided-task-configurations.ts

```diff
@@ -111,6 +121,13 @@ export class ProvidedTaskConfigurations {
         return matchedTask;
     }
 
+    /** checks if the config is a detected / contributed task */
+    isDetectedTask(task: TaskCustomization | TaskConfiguration | TaskCustomization): task is ContributedTaskConfiguration {
```

Comment:
We should be consistent with naming. Either call them provided, detected or contributed. But please not mix.